packages/core-flows/src/region/steps/upsert-and-replace-region-payment-providers.ts

```diff
@@ -0,0 +1,284 @@
+import {
+  ModuleRegistrationName,
+  Modules,
+  RemoteLink,
+} from "@medusajs/modules-sdk"
+import {
+  FilterableRegionProps,
+  IPaymentModuleService,
+  IRegionModuleService,
+  RemoteQueryFunction,
+} from "@medusajs/types"
+import { createStep, StepResponse } from "@medusajs/workflows-sdk"
+import {
+  arrayDifference,
+  ContainerRegistrationKeys,
+  MedusaError,
+  promiseAll,
+  remoteQueryObjectFromString,
+} from "@medusajs/utils"
+
+export interface UpsertAndReplaceRegionPaymentProvidersStepInput {
```

Comment:
suggestion: Would it make sense to simplify the input to:
```ts
interface RegionPPInput {
  id: string
  payment_providers: string[]
}

interface StepInput {
  input: RegionPPInput[]
}
```

Can we achieve what we want to?

Reply:
Yes we can definitely, in the case the step becomes again just a setter kind of and the region validation/querying falls back in the hands of the consumer, which is not bad i believe